src/route-recognizer.js

```diff
@@ -135,13 +142,15 @@ export class RouteRecognizer {
         continue;
       }
 
-      output += '/';
       let segmentValue = segment.generate(routeParams, consumed);
-      if (segmentValue === null || segmentValue === undefined) {
-        throw new Error(`A value is required for route parameter '${segment.name}' in route '${name}'.`);
+      if (segmentValue == null) {
```

Comment:
Does this pass the lint task? Sorry, I can't remember if our rules require strict equality. #lazy